diff -Naur mysql-connector-c++-1.1.0/cppconn/CMakeLists.txt mysql-connector-c++-1.1.0.oden/cppconn/CMakeLists.txt
--- mysql-connector-c++-1.1.0/cppconn/CMakeLists.txt	2010-09-10 11:47:47.000000000 +0200
+++ mysql-connector-c++-1.1.0.oden/cppconn/CMakeLists.txt	2010-12-29 13:27:19.023849777 +0100
@@ -69,4 +69,4 @@
 		sqlstring.h
 		warning.h)
 
-INSTALL(FILES ${MYSQLCPPCONN_INSTALL_HEADERS} DESTINATION include/cppconn)
+INSTALL(FILES ${MYSQLCPPCONN_INSTALL_HEADERS} DESTINATION include/mysql-connector/cppconn)
diff -Naur mysql-connector-c++-1.1.0/driver/CMakeLists.txt mysql-connector-c++-1.1.0.oden/driver/CMakeLists.txt
--- mysql-connector-c++-1.1.0/driver/CMakeLists.txt	2010-09-10 11:47:47.000000000 +0200
+++ mysql-connector-c++-1.1.0.oden/driver/CMakeLists.txt	2010-12-29 13:27:19.023849777 +0100
@@ -23,6 +23,9 @@
 
 LINK_DIRECTORIES(${MYSQLCPPCONN_BOOST_LIBRARY_DIRS})
 
+IF(NOT DEFINED CMAKE_INSTALL_LIB_DIR)
+  SET(CMAKE_INSTALL_LIB_DIR lib)
+ENDIF(NOT DEFINED CMAKE_INSTALL_LIB_DIR)
 
 INCLUDE_DIRECTORIES(${MYSQLCPPCONN_SOURCE_DIR})
 INCLUDE_DIRECTORIES(${MYSQLCPPCONN_SOURCE_DIR}/cppconn)
@@ -271,8 +274,8 @@
 	)
 ELSE(WIN32)
 	INSTALL(TARGETS mysqlcppconn mysqlcppconn-static
-		LIBRARY DESTINATION lib
-		ARCHIVE DESTINATION lib
+		LIBRARY DESTINATION ${CMAKE_INSTALL_LIB_DIR}
+		ARCHIVE DESTINATION ${CMAKE_INSTALL_LIB_DIR}
 	)
 ENDIF(WIN32)
 
@@ -282,7 +285,7 @@
 		mysql_connection.h
 		mysql_driver.h)
 
-INSTALL(FILES ${MYSQLCPPCONN_SPECIFIC_INSTALL_HEADERS} DESTINATION include)
+INSTALL(FILES ${MYSQLCPPCONN_SPECIFIC_INSTALL_HEADERS} DESTINATION include/mysql-connector)
 
 
 MESSAGE(STATUS "Configuring driver")
