diff -Naurp mysql-connector-c++-1.1.0/driver/mysql_art_resultset.cpp mysql-connector-c++-1.1.0.oden/driver/mysql_art_resultset.cpp
--- mysql-connector-c++-1.1.0/driver/mysql_art_resultset.cpp	2010-09-10 11:47:47.000000000 +0200
+++ mysql-connector-c++-1.1.0.oden/driver/mysql_art_resultset.cpp	2010-12-29 13:30:15.254005988 +0100
@@ -23,6 +23,7 @@
 */
 
 #include <sstream>
+#include <stdio.h>
 #include <math.h>
 #include <stdio.h>
 
diff -Naurp mysql-connector-c++-1.1.0/driver/mysql_resultbind.cpp mysql-connector-c++-1.1.0.oden/driver/mysql_resultbind.cpp
--- mysql-connector-c++-1.1.0/driver/mysql_resultbind.cpp	2010-09-10 11:47:47.000000000 +0200
+++ mysql-connector-c++-1.1.0.oden/driver/mysql_resultbind.cpp	2010-12-29 13:30:15.254005988 +0100
@@ -32,6 +32,7 @@
 #include "nativeapi/native_resultset_wrapper.h"
 
 #include <string.h>
+#include <stdio.h>
 
 namespace sql
 {
diff -Naurp mysql-connector-c++-1.1.0/test/unit/classes/resultsetmetadata.cpp mysql-connector-c++-1.1.0.oden/test/unit/classes/resultsetmetadata.cpp
--- mysql-connector-c++-1.1.0/test/unit/classes/resultsetmetadata.cpp	2010-09-10 11:47:47.000000000 +0200
+++ mysql-connector-c++-1.1.0.oden/test/unit/classes/resultsetmetadata.cpp	2010-12-29 13:30:15.353213722 +0100
@@ -27,6 +27,7 @@
 #include "resultsetmetadata.h"
 #include <sstream>
 #include <stdlib.h>
+#include <stdio.h>
 #include <cppconn/resultset.h>
 #include <cppconn/datatype.h>
 #include <cppconn/prepared_statement.h>
